packages/blade/src/components/ActionList/ActionListFooter.tsx

```diff
@@ -0,0 +1,142 @@
+
+import React from 'react';
+import styled from 'styled-components';
+import { componentIds } from './componentIds';
+import { getActionListFooterRole } from './getA11yRoles';
+import Box from '~components/Box';
+import type { IconComponent } from '~components/Icons';
+import { useDropdown } from '~components/Dropdown/useDropdown';
+import {
+  isReactNative,
+  makeSize,
+  makeAccessible,
+  MetaConstants,
+  metaAttribute,
+  isValidAllowedChildren,
+} from '~utils';
+import type { WithComponentId } from '~utils';
+import { Text } from '~components/Typography';
+
+type ActionListFooterProps = {
+  title?: string;
+  leading?: React.ReactNode;
+  trailing?: React.ReactNode;
+};
+
+const StyledActionListFooter = styled(Box)((props) => {
+  return {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    padding: `${makeSize(props.theme.spacing[3])} ${makeSize(props.theme.spacing[5])}`,
+    backgroundColor: props.theme.colors.brand.gray.a50.lowContrast,
+    margin: `${makeSize(props.theme.spacing[3])} -${makeSize(props.theme.spacing[3])} -${makeSize(
+      props.theme.spacing[3],
+    )}`,
+  };
+});
+
+const ActionListFooter: WithComponentId<ActionListFooterProps> = (props): JSX.Element => {
+  const footerRef = React.useRef<HTMLDivElement | null>(null);
+  const {
+    setShouldIgnoreBlur,
+    setHasFooterAction,
+    onTriggerKeydown,
+    activeIndex,
+    setIsOpen,
+    selectionType,
+  } = useDropdown();
+
+  React.useEffect(() => {
+    React.Children.map(props.leading, (child) => {
+      if (!isValidAllowedChildren(child, componentIds.ActionListFooterIcon)) {
+        throw new Error(
+          `[ActionListFooter]: Only ${componentIds.ActionListFooterIcon} is allowed in leading prop`,
+        );
+      }
+    });
+  }, [props.leading]);
+
+  React.useEffect(() => {
+    // We only need this in web to handle some keydown events
+    if (!isReactNative() && footerRef.current?.querySelector('button, a')) {
+      setHasFooterAction(true);
+    }
+  }, [setHasFooterAction, props.trailing]);
+
+  const isOnlyActionButton = !props.title && !props.leading && props.trailing;
+
+  return (
+    <StyledActionListFooter
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      ref={footerRef as any}
+      // @ts-ignore: Ignoring because the TS fails for React Native and works for web
+      onMouseDown={() => {
+        if (selectionType === 'multiple') {
+          setShouldIgnoreBlur(true);
+        }
+      }}
+      // @ts-ignore: Ignoring because the TS fails for React Native and works for web
+      onKeyDown={(e) => {
+        const nativeEvent = e.nativeEvent;
+        const shouldIgnoreDropdownKeydown =
+          (nativeEvent.key === ' ' || nativeEvent.key === 'Enter') && activeIndex < 0;
+        // We ignore the selection keydowns on footer to let users click on items on the footer
+        if (shouldIgnoreDropdownKeydown) {
+          if (selectionType === 'single') {
```

Comment:
Can be done with 1 if statement? 
shouldIgnoreDropdownKeydown && selectionType == single

Reply:
Nope its not the same right?

we always want to run that `return` when `shouldIgnoreDropdownKeydown` is true. But we just also want to run `setIsOpen(false` when `selectionType === single` is true.

With `&&` it will only run that block when both of it is true.